src/program/lwaftr/bench/bench.lua

```diff
@@ -58,19 +77,32 @@ function run(args)
       csv:add_app('sinkv6', { 'input' }, { input=opts.hydra and 'encap' or 'Encap.' })
       csv:activate()
    end
-   for _,pid in ipairs(app.configuration.apps['leader'].arg.follower_pids) do
-      -- The worker will be fed its configuration by the
-      -- leader, but we don't know when that will all be ready.
-      -- Just retry if this doesn't succeed.
-      local function start_sampling()
-	 if not pcall(start_sampling_for_pid, pid) then
-	    io.stderr:write("Waiting on follower "..pid.." to start "..
-			       "before recording statistics...\n")
-	    timer.activate(timer.new('retry_csv', start_sampling, 1e9))
-	 end
+   
+   -- We don't know the followers PIDs until the leader starts them. There may
+   -- be additional followers started as instances that have been added or
+   -- removed during the running. We need to look for the PIDs and start
+   -- sampling for each new PID we find.
+   local function start_sampling()
+      local followers = {}
+      local function cycle()
+         local new_followers = find_followers()
+         for pid, _ in pairs(new_followers) do
+            if followers[pid] == nil then
+               if not pcall(start_sampling_for_pid, pid) then
+                  new_followers[pid] = nil
+                  --io.stderr:write("Waiting on follower "..pid.." to start "..
+                  --   "before recording statistics...\n")
+               end
+            end
+         end
+         followers = new_followers
+         
+         timer.activate(timer.new('start_sampling', cycle, 1e9))
       end
-      timer.activate(timer.new('spawn_csv_stats', start_sampling, 10e6))
+      timer.activate(timer.new('start_sampling', cycle, 1e9))
```

Comment:
Here you can add a fourth argument, 'repeating', to make a repeating timer.  But how do we want to treat statistics for multiple workers?  Sum or list separately?  A question for a followup, perhaps.

Reply:
I've changed it to be repeating.

I'm not sure if how to handle multiple sets of statistics. I'm currently just listing them separately, I don't know if that should change. It probably shouldn't print the header multiple times which it does. I can disable that for now and if they want to be summed or something other than just listed individually for each instance maybe that can come later.